nano_graphrag/storage/asyncpg.py

```diff
@@ -0,0 +1,100 @@
+from nano_graphrag._storage import BaseVectorStorage
+import asyncpg
+import asyncio
+from contextlib import asynccontextmanager
+from nano_graphrag._utils import logger
+from pgvector.asyncpg import register_vector
+from nano_graphrag.graphrag import always_get_an_event_loop
+import numpy as np
+import json
+
+import nest_asyncio
+nest_asyncio.apply()
```

Comment:
The reason is that we seem to lack an asynchronous initialization function. The entire function runs in an asynchronous environment, and in order to run in a nested asynchronous environment, I used nest_asyncio.

Reply:
Yeah... it could be problematic I think. Do we have to use nest-async to run pg-vector storage?

Reply:
I need to ensure that the plugin is created correctly, along with the corresponding table. Since the asyncpg library only supports asynchronous operations, I need to obtain a loop for initialization.